internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -482,9 +473,23 @@ func (r *Reconciler) reconcileMachineDeployments(ctx context.Context, s *scope.S
 }
 
 // createMachineDeployment creates a MachineDeployment and the corresponding Templates.
-func (r *Reconciler) createMachineDeployment(ctx context.Context, cluster *clusterv1.Cluster, md *scope.MachineDeploymentState) error {
-	log := tlog.LoggerFrom(ctx).WithMachineDeployment(md.Object)
+func (r *Reconciler) createMachineDeployment(ctx context.Context, s *scope.Scope, md *scope.MachineDeploymentState) error {
+	// Do not create the MachineDeployment if it is marked as pending create.
+	// This will also block MHC creation because creating the MHC without the corresponding
+	// MachineDeployment is unnecessary.
+	mdTopologyName, ok := md.Object.Labels[clusterv1.ClusterTopologyMachineDeploymentNameLabel]
+	if !ok || mdTopologyName == "" {
+		// Note: This is only an additional safety check and should not happen. The label will always be added when computing
+		// the desired MachineDeployment.
+		return fmt.Errorf("new MachineDeployment is missing the %q label", clusterv1.ClusterTopologyMachineDeploymentNameLabel)
```

Comment:
```suggestion
		return errors.Errorf("new MachineDeployment is missing the %q label", clusterv1.ClusterTopologyMachineDeploymentNameLabel)
```
nit

Reply:
Fixed this. 
In the future should we always use `errors.Errorf` over `fmt.Errorf`? Is it for consistency or is there anther reasons too? (consistency is a good enough reason just curious to know if there is more to this)